src/clj/fluree/db/ledger/json_ld.cljc

```diff
@@ -90,22 +96,155 @@
+          signed-commit        (if did
+                                 (<? (cred/generate jld-commit private (:id did)))
+                                 jld-commit)
+          commit-res           (<? (connection/-c-write conn alias signed-commit))
+          commit**             (commit-data/update-commit-address commit* (:address commit-res))]
+      {:commit-map    commit**
+       :commit-jsonld jld-commit
+       :write-result  commit-res})))
+
+(defn push-commit
+  [conn {:keys [state] :as _ledger} {:keys [commit-map commit-jsonld write-result]}]
+  (nameservice/push! conn (assoc commit-map
+                                 :meta write-result
+                                 :json-ld commit-jsonld
+                                 :ledger-state state)))
+
+(defn do-commit+push
+  "Writes commit and pushes, kicks off indexing if necessary."
+  [{:keys [conn alias] :as ledger}
+   {:keys [branch], new-commit :commit, :as db}
+   keypair
+   index-files-ch]
+  (go-try
+    (log/debug "do-commit+push new-commit:" new-commit)
+    (let [ledger-commit (current-commit ledger branch)
+
+          {:keys [commit-map write-result] :as commit-write-map}
+          (<? (write-commit conn alias keypair new-commit))
+
+          db*      (assoc db :commit commit-map)
+          db**     (if (new-t? ledger-commit new-commit)
+                     (commit-data/add-commit-flakes db* (:prev-commit db))
+                     db*)
+          db***    (commit-update ledger branch db** index-files-ch)
+          push-res (<? (push-commit conn ledger commit-write-map))]
+      {:commit-res write-result
+       :push-res   push-res
+       :db         db***})))
+
+(defn write-transactions!
+  [conn {:keys [alias] :as _ledger} staged]
+  (go-try
+    (loop [[[txn author-did annotation] & r] staged
+           results                []]
+      (if txn
+        (let [{txn-id :address} (<? (connection/-txn-write conn alias txn))]
+          (recur r (conj results [txn-id author-did annotation])))
+        results))))
+
 (defn commit!
-  [ledger db opts]
-  (let [{:keys [branch] :as opts*}
-        (normalize-opts opts)
-        {:keys [t] :as db*} (or db (ledger/-db ledger branch))
-        committed-t                (ledger/latest-commit-t ledger branch)]
-    (if (= t (flake/next-t committed-t))
-      (jld-commit/commit ledger db* opts*)
-      (throw (ex-info (str "Cannot commit db, as committed 't' value of: " committed-t
-                           " is no longer consistent with staged db 't' value of: " t ".")
-                      {:status 400 :error :db/invalid-commit})))))
+  "Finds all uncommitted transactions and wraps them in a Commit document as the subject
+  of a VerifiableCredential. Persists according to the :ledger :conn :method and
+  returns a db with an updated :commit."
+  [{:keys [alias conn] :as ledger} {:keys [t stats commit] :as db} opts]
+  (go-try
+    (let [{:keys [did message tag file-data? index-files-ch] :as opts*}
```

Comment:
Does anything actually supply an `index-files-ch` opt yet? Is that something that `server` will supply?

Reply:
Also, a nitpick: `index-files-ch` may be a slight misnomer if the FileStore isn't being used. Maybe `index-node-ch`?

Reply:
fluree/server supplies it, and it is used to distributed the index file notifications across the network.